### PR TITLE
Improve share modal behavior

### PR DESCRIPTION
**Pull request by Kadrian**
- Close after successful share (was more of a bug)
- Indicate loading state
- Disable button as long as no user groups have changed.

### CODE REVIEW

File: frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx [hunk 1]
```diff
@@ -110,28 +117,32 @@ const ShareProjectItemModal = ({ item, onClose }: PropsT) => {
       ? t("sharePreviousQueryModal.unshare")
       : t("common.share");
 
+  const buttonDisabled =
+    JSON.stringify(initialUserGroupsValue) === JSON.stringify(userGroupsValue);
```
Comment (awildturtok) [inline frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx:121]: This seems rather hacky to me, relying on the serialization of the state object? Wouldn't it be preferable to use an elementwise comparison instead (likely also much more efficient)

**Review by awildturtok — COMMENTED** [review]
It seems array equality is a [contentious topic](https://stackoverflow.com/questions/7837456/how-to-compare-arrays-in-javascript) in the JS community. I'd personally still rely on a less magic way, or at least encode it explicitly in a helper method (which can easily be plugged to use "better" variants should they be released). But it's not a mjaor problem.

File: frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx (hunk 1 above)
Comment (Kadrian) [reply frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx:121]: I understand your point, but user groups are short arrays, this is efficient enough, so I'd rather avoid micro optimizations here

Comment (awildturtok) [reply frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx:121]: My point isn't as much about optimization than about misappropriating serialization for something as basic as equality checks. But if you'd rather keep it I'm fine with it.

Comment (Kadrian) [reply frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx:121]: Agreed that it feels weird until you've seen it lots of times in JS code. It's more common than you'd think: https://stackoverflow.com/a/1144249/921486

**Review by awildturtok — APPROVED** [review]

**Pull request merged**